### PR TITLE
apr: add apr/1.7.0 recipe

### PR DESCRIPTION
**Pull request by madebr**
Specify library name and version:  **apr/1.7.0**

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/wiki#how-to-submit-a-pull-request) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

File: recipes/apr/all/conanfile.py [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+            tools.rmdir(os.path.join(self.package_folder, "build-1"))
+            tools.rmdir(os.path.join(self.package_folder, "lib", "pkgconfig"))
+
+    def package_info(self):
+        self.cpp_info.names["pkg_config"] = "apr-1"
+        self.cpp_info.libs = ["apr-1"]
+        if not self.options.shared:
+            self.cpp_info.defines = ["APR_DECLARE_STATIC"]
+            if self.settings.os == "Linux":
+                self.cpp_info.system_libs = ["dl", "pthread"]
```
Comment (uilianries) [inline recipes/apr/all/conanfile.py:106]:
```suggestion
        if self.settings.os == "Linux":
            self.cpp_info.system_libs = ["dl", "pthread"]
```

**Comment by fulara** [comment]
Testing recipe.
apr_rules.mk generated by your build remembers the path where it was built into.
I think this is a problem so for me it looks like this:
```
apr_builddir=/var/build/bamboo/agent-home/xml-data/build-dir/CIR-APR2-JOB1/.conan-standalone/bamboo/.conan/data/apr/1.7.0/_/_/package/9fdb4217a0bb5bac441d7e17705c9172eeeb6cfe/bin/build-1
apr_builders=/var/build/bamboo/agent-home/xml-data/build-dir/CIR-APR2-JOB1/.conan-standalone/bamboo/.conan/data/apr/1.7.0/_/_/package/9fdb4217a0bb5bac441d7e17705c9172eeeb6cfe/bin/build-1
top_builddir=/var/build/bamboo/agent-home/xml-data/build-dir/CIR-APR2-JOB1/.conan-standalone/bamboo/.conan/data/apr/1.7.0/_/_/package/9fdb4217a0bb5bac441d7e17705c9172eeeb6cfe/bin/build-1
```
but my conan current $CONAN_USER_HOME is:
`/var/build/fulara/conan-standalone`

I think some of the things do peek into that directory.

Keep in midn that these variables are actually sed' during the build of the apr, so you cannot modify them at the beginning of `source` step.

What I ended doing in my package was something along these lines:
```
    def package(self):
        # .. installing code here
        self.run("sed -i 's@apr_builddir=.*@apr_builddir=${APR_PACKAGE_DIR_CONAN}/build-1@g' " + os.path.join(self.package_folder, "build-1", "apr_rules.mk"))
        self.run("sed -i 's@apr_builders=.*@apr_builders=${APR_PACKAGE_DIR_CONAN}/build-1@g' " + os.path.join(self.package_folder, "build-1", "apr_rules.mk"))
        self.run("sed -i 's@top_builddir=.*@top_builddir=${APR_PACKAGE_DIR_CONAN}/build-1@g' " + os.path.join(self.package_folder, "build-1", "apr_rules.mk"))
```
and in the `package_info` I am exporting variable, - you can use the APR_ROOT in your case.

**Comment by fulara** [comment]
I think on top of problem mentioned in the above error I am seeing this:
```
error	27-Apr-2020 21:41:56		ConanException: Error 1 while executing source_subfolder/configure '--with-apr=/var/build/bamboo/agent-home/xml-data/build-dir/CIR-APRUTIL2-JOB1/.conan-standalone/bamboo/.conan/data/apr/1.7.0/_/_/package/9fdb4217a0bb5bac441d7e17705c9172eeeb6cfe' '--with-crypto' '--with-expat=/var/build/bamboo/agent-home/xml-data/build-dir/cir-aprutil2-job1 ...
```
Note that paths.
once its lupper case CIR-APRUTIL2-JOB1 and once its lower case.
Usage of unix_path is dooming me here.  
on top of that it seems to be an inconsistent usage of unix_path. (apr is not passed the unix_path but  --with-expat is. )

File: recipes/apr/all/conanfile.py (hunk 1 above)
Comment (madebr) [reply recipes/apr/all/conanfile.py:106]: This is correct.
Shared libraries don't need any extra libraries.
The headers also do not expose any symbols other then the `apr_*` ones.

**Comment by madebr** [comment]
@fulara 
I've added a patch to `apr` and `apr-util`.
Can you recheck?

File: recipes/apr/all/conanfile.py [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+        tools.get(**self.conan_data["sources"][self.version])
+        os.rename("{}-{}".format(self.name, self.version), self._source_subfolder)
+
+    def _configure_cmake(self):
+        if self._cmake:
+            return self._cmake
+        self._cmake = CMake(self)
+        self._cmake.definitions["INSTALL_PDB"] = False
+        self._cmake.definitions["APR_BUILD_TESTAPR"] = False
+        # self._cmake.definitions["APR_INSTALL_PRIVATE_H"] = True
```
Comment (SSE4) [inline recipes/apr/all/conanfile.py:57]: remove dead code pls

Comment (madebr) [inline recipes/apr/all/conanfile.py:57]:
```suggestion
```

File: recipes/apr/all/conanfile.py (hunk 1 above)
Comment (uilianries) [reply recipes/apr/all/conanfile.py:114]: okay, fine for this case, but remember we already discussed about it few months ago. checking for it increases the recipes complexity, also exposing those libraries are harmless and the user will need them anyway.

**Pull request merged**